src/map/entities/battleentity.cpp

```diff
@@ -712,7 +712,8 @@ uint16 CBattleEntity::ATT()
     {
         ATT += this->GetSkill(SKILL_AUTOMATON_MELEE);
     }
-    return ATT + (ATT * ATTP / 100) + std::min<int16>((ATT * m_modStat[Mod::FOOD_ATTP] / 100), m_modStat[Mod::FOOD_ATT_CAP]);
+    // use max to prevent underflow
+    return std::max(0, ATT + (ATT * ATTP / 100) + std::min<int16>((ATT * m_modStat[Mod::FOOD_ATTP] / 100), m_modStat[Mod::FOOD_ATT_CAP]));
```

Comment:
Shouldn't these clamp to a lower bounds of `1` to avoid divisions by zero?

Reply:
> Shouldn't these clamp to a lower bounds of `1` to avoid divisions by zero?

I am not sure, I took the zero lower bound from the clamping and max of other existing stats like ACC and DEF. I am not sure if on retail you can get to zero?

Reply:
Ok, after discussion with others, I set the lower bound of def and eva to 1 and the lower bound of acc, racc, att, and ratt to 0 (as these can be zero according to for example checkparams).

Reply:
/checkparam can report 0 ratt/racc if you have no ranged weapon, that's probably why. I don't think it can ever report 0 for your melee weapon (Specifically to avoid dividing by zero)

Reply:
> /checkparam can report 0 ratt/racc if you have no ranged weapon, that's probably why. I don't think it can ever report 0 for your melee weapon (Specifically to avoid dividing by zero)

Yep, I have changed ATT to have lower bound of 1.